src/ert/_c_wrappers/job_queue/ert_script.py

```diff
@@ -81,34 +57,33 @@ def initializeAndRun(
                 arguments.append(None)
 
         try:
-            return self.run(*arguments)
+            self.run_status.start()
+            msg = self.run(*arguments)
+            self.run_status.finish(msg)
         except AttributeError as e:
+            error_msg = e
             if not hasattr(self, "run"):
-                self.__failed = True
-                return (
-                    f"Script '{self.__class__.__name__}' "
-                    "has not implemented a 'run' function"
-                )
-            self.outputStackTrace(e)
-            return None
+                error_msg = "No 'run' function implemented"
+            self.output_stack_trace(error=error_msg)
         except KeyboardInterrupt:
-            return f"Script '{self.__class__.__name__}' cancelled (CTRL+C)"
+            error_msg = "Script cancelled (CTRL+C)"
```

Comment:
Seems like mypy is not agreeing with what you give to the `error` argument here as you seem to give it str in one setting and ArgumentError in the other. Should probably give it the union type in the `outputStackTrace` function or explicitly convert the error to a string.